src/bpftrace.cpp

```diff
@@ -809,7 +838,8 @@ int BPFtrace::run(Output &out, BpfBytecode bytecode)
     if (err)
       return err;
   } else {
-    bool should_drain = num_begin_end_attached > 0 &&
+    bool should_drain = (num_begin_end_attached > 0 ||
+                         !benchmark_results.empty()) &&
```

Comment:
Nit: Maybe this is slightly more grokable
```suggestion
    bool should_drain = (num_begin_end_attached > 0 ||
                        run_benchmarks_) &&
```

Reply:
Changed.